toolkit/partner-chains-cli/src/deregister/mod.rs

```diff
@@ -1,100 +1,57 @@
 #[cfg(test)]
 mod tests;
 
+use crate::cardano_key::{get_mc_pkey_from_file, get_mc_pubkey_from_file};
 use crate::config::config_fields::{
 	CARDANO_COLD_VERIFICATION_KEY_FILE, CARDANO_PAYMENT_SIGNING_KEY_FILE,
 };
-use crate::config::{
-	get_cardano_network_from_file, ChainConfig, CHAIN_CONFIG_FILE_PATH, PC_CONTRACTS_CLI_PATH,
-};
+use crate::config::CHAIN_CONFIG_FILE_PATH;
 use crate::io::IOContext;
-use crate::pc_contracts_cli_resources::{
-	establish_pc_contracts_cli_configuration, PcContractsCliResources,
-};
-use crate::{smart_contracts, CmdRun};
-use anyhow::{anyhow, Context};
-use sidechain_domain::NetworkType;
+use crate::pc_contracts_cli_resources::establish_pc_contracts_cli_configuration;
+use crate::CmdRun;
+use anyhow::anyhow;
+use partner_chains_cardano_offchain::register::Deregister;
 
 #[derive(Debug, clap::Parser)]
 pub struct DeregisterCmd;
 
 impl CmdRun for DeregisterCmd {
 	fn run<C: IOContext>(&self, context: &C) -> anyhow::Result<()> {
-		let (chain_config, cardano_network) = read_chain_config_file(context)?;
+		let chain_config = read_chain_config_file(context)?;
 		context.print(
 		  &format!("This wizard will remove the specified candidate from the committee candidates based on the following chain parameters:\n{}.\nCommittee Candidate Validator Address is '{}'\n",
           &chain_config.chain_parameters,
           &chain_config.cardano_addresses.committee_candidates_address)
     	);
-		let (payment_signing_key_path, cold_vkey) = get_cardano_keys(context)?;
+		context.print("Payment signing key and verification key of cold key used for registration are required to deregister.");
+		let payment_signing_key_path =
+			CARDANO_PAYMENT_SIGNING_KEY_FILE.prompt_with_default_from_file_and_save(context);
+		let payment_signing_key = get_mc_pkey_from_file(&payment_signing_key_path, context)?;
+		let cold_vkey_path =
+			CARDANO_COLD_VERIFICATION_KEY_FILE.prompt_with_default_from_file_and_save(context);
+		let stake_ownership_pub_key = get_mc_pubkey_from_file(&cold_vkey_path, context)?;
 		let pc_contracts_cli_resources = establish_pc_contracts_cli_configuration(context)?;
-
-		let command = build_command(
-			cardano_network,
-			cold_vkey,
-			chain_config,
-			pc_contracts_cli_resources,
-			payment_signing_key_path,
-		);
-		let output = context.run_command(&command).context("Deregistration failed")?;
-		Ok(context.print(&format!("Deregistration successful: {}", output)))
+		let offchain = context.offchain_impl(&pc_contracts_cli_resources.ogmios)?;
+
+		let runtime = tokio::runtime::Runtime::new().map_err(|e| anyhow::anyhow!(e))?;
+		runtime
+			.block_on(offchain.deregister(
+				chain_config.chain_parameters.genesis_utxo,
+				payment_signing_key,
+				stake_ownership_pub_key,
+			))
+			.map_err(|e| anyhow::anyhow!("Candidate registration failed: {e:?}!"))?;
+
+		Ok(())
```

Comment:
Cosmetic: is it required? Isn't removing `?;` two lines above enough?

Reply:
Unfortunately it is, as `deregister` returns `Option<McTxHash>` and `run` has to return `Result<()>`